libraries/freertos_plus/standard/https/include/iot_https_client.h

```diff
@@ -452,4 +687,44 @@ IotHttpsReturnCode_t IotHttpsClient_ReadContentLength( IotHttpsResponseHandle_t
 IotHttpsReturnCode_t IotHttpsClient_ReadHeader(IotHttpsResponseHandle_t respHandle, char *pName, char *pValue, uint32_t len);
 /* @[declare_https_client_readheader] */
 
+/**
+ * @brief Read the HTTPS response body from the network.
+ * 
+ * This is intended to be used with an asynchronous response, this is to be invoked during the 
+ * #IotHttpsClientCallbacks_t.readReadyCallback to read data directly from the network into pBuf.
+ * <b> Example Asynchronous Code </b>
+ * @code{c}
+ * void applicationDefined_readReadyCallback(void * pPrivData, IotHttpsRequestHandle_t handle, IotHttpsReturnCode_t rc, uint16_t status)
+ * {
+ *      ...
+ *      char * myBuf = STORE_ADDRESS;
+ *      uint32_t len = STORE_READ_SIZE;
+ *      IotHttpsClient_ReadResponseBody(handle, myBuf, &len);
+ *      ...
+ * }
+ * @endcode
+ * 
+ * For a syncrhonous response, to retrieve the response body applications must directly refer to the 
+ * #pSyncInfo_t.pRespData configured in #IotHttpsRequestInfo_t.pSyncInfo_t. Otherwise this function will return an 
+ * #IOT_HTTPS_INVALID_PARAMETER error code. This function is intended to read the response entity body from the network 
+ * and the synchronous response process handles all of that in @ref https_client_function_sendsync.
```

Comment:
The more I see this, the more I am concerned that this is confusing to app developers. Reading through this, its not clear to me how to read response body from a sync request.

Reply:
They need to go to memory location they configured before calling IotHttpsClient_SendSync(). 
```
For a syncrhonous response, to retrieve the response body applications must directly refer to the 
 * #IotHttpsResponseInfo_t.pSyncInfo_t.pBody.
```